core/services/gas_updater.go

```diff
@@ -57,91 +70,292 @@ type gasUpdater struct {
 	// it too early results in an empty block.
 	blockDelay int64
 	percentile int
+	mb         *utils.Mailbox
+	wg         *sync.WaitGroup
+	ctx        context.Context
+	ctxCancel  context.CancelFunc
+
+	logger *logger.Logger
+}
+
+type blockInternal struct {
```

Comment:
I think we can pretty easily put the `gas_update.go` and `gas_updater_test.go` into their own gasupdater package to limit the baggage that comes in with services. Then we can also break this into its own `block.go` file

Reply:
Do we really want to do that? I'm nervous due to our package fragmentation causing so many import cycles before

Reply:
I think we do, it enforces encapsulation of this logic and makes the dependencies explicit. There's always a way to break a cycle by pulling out a common package if you need to. The cycle mess we had before, as far as I know, was primarily due to cltest importing everything, but we have a pattern of doing blackbox testing with a _test package to resolve that.